spec/javascripts/govuk_publishing_components/lib/govspeak/youtube-link-enhancement-spec.js

```diff
@@ -1,64 +1,117 @@
+    		'essential': true,
+    		'settings': true,
+    		'usage': true
+    	})
+    	window.GOVUK.cookie('cookie_policy', null)
+    })
 
-    expect($element.find('.media-player .video').length).toBe(1)
-    expect($element.find($toReplace).length).toBe(0)
-  })
+    it('replaces a link and it\'s container with a media-player embed', function () {
+    	var $element = $('<div><p><a href="https://www.youtube.com/watch?v=0XpAtr24uUQ">Agile at GDS</a></p></div>')
+    	var $toReplace = $element.find('p, a')
 
-  it('doesn\'t replace non Youtube links', function () {
-    var $element = $('<div><p><a href="https://www.gov.uk/">GOV.UK</a></p></div>')
-    var $toReplace = $element.find('p, a')
+    	var enhancement = new GOVUK.GovspeakYoutubeLinkEnhancement($element)
+    	enhancement.init()
+    	expect($element.find('.media-player .video').length).toBe(1)
+    	expect($element.find($toReplace).length).toBe(0)
+    })
 
-    var enhancement = new GOVUK.GovspeakYoutubeLinkEnhancement($element)
-    enhancement.init()
+    it('doesn\'t replace non Youtube links', function () {
+    	var $element = $('<div><p><a href="https://www.gov.uk/">GOV.UK</a></p></div>')
+    	var $toReplace = $element.find('p, a')
 
-    expect($element.find('.media-player .video').length).toBe(0)
-    expect($element.find($toReplace).length).toBe(2)
-  })
+    	var enhancement = new GOVUK.GovspeakYoutubeLinkEnhancement($element)
+    	enhancement.init()
 
-  it('doesn\'t replace links marked not to embed', function () {
-    var $element = $('<div><p><a href="https://www.youtube.com/watch?v=0XpAtr24uUQ" data-youtube-player="off">Agile at GDS</a></p></div>')
-    var $toReplace = $element.find('p, a')
+    	expect($element.find('.media-player .video').length).toBe(0)
+    	expect($element.find($toReplace).length).toBe(2)
+    })
 
-    var enhancement = new GOVUK.GovspeakYoutubeLinkEnhancement($element)
-    enhancement.init()
+    it('doesn\'t replace links marked not to embed', function () {
+    	var $element = $('<div><p><a href="https://www.youtube.com/watch?v=0XpAtr24uUQ" data-youtube-player="off">Agile at GDS</a></p></div>')
+    	var $toReplace = $element.find('p, a')
 
-    expect($element.find('.media-player .video').length).toBe(0)
-    expect($element.find($toReplace).length).toBe(2)
-  })
+    	var enhancement = new GOVUK.GovspeakYoutubeLinkEnhancement($element)
+    	enhancement.init()
 
-  describe('parseVideoId', function () {
-    it('returns an id for a youtube.com video URL', function () {
-      var url = 'https://www.youtube.com/watch?v=0XpAtr24uUQ'
-      var id = GOVUK.GovspeakYoutubeLinkEnhancement.parseVideoId(url)
+    	expect($element.find('.media-player .video').length).toBe(0)
+    	expect($element.find($toReplace).length).toBe(2)
+    })
 
-      expect(id).toEqual('0XpAtr24uUQ')
+    it('doesn\'t replace links when campaign cookies are turned off', function () {
+    	window.GOVUK.cookie('cookie_policy', JSON.stringify({
+    		campaigns: false
```

Comment:
I'm not 100% sure this is working as we expect. I *think* when I looked earlier, this was returning `"{\"campaigns\":false}"` as the cookie value, which meant JSON.parse was failing. This meant `YoutubeLinkEnhancement.campaignCookies` was returning `true` (correct) but only because the value of the cookie for campaigns was `undefined`. So it's not quite testing what happens when the campaigns value is true.